src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/ExternallyInteractable.java

```diff
@@ -0,0 +1,116 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
+import org.bukkit.Location;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Implement this interface for any {@link SlimefunItem} to provide methods for
+ * external code to 'interact' with the item when placed as a block in the world.
+ *
+ * @author Sefiraat
+ *
+ */
+public interface ExternallyInteractable {
+
+    /**
+     * This method should be used by the implementing class to fulfill the actions needed
+     * when being interacted with returning the result of the interaction.
+     *
+     * @param location
+     *                 The {@link Location} of the Block being targeted for the interaction.
+     *
+     * @return The {@link InteractionResult} denoting the result of the interaction.
+     */
+    @Nonnull InteractionResult onInteract(@Nonnull Location location);
+
+    /**
+     * This class represents the result of an interaction on an {@link ExternallyInteractable} item.
+     */
+    class InteractionResult {
+
+        private final boolean interactionSuccessful;
+        private @Nullable String resultMessage;
+        private final @Nonnull Set<ItemStack> resultItems = new HashSet<>();
```

Comment:
I don't want this to turn into a huge class with 10 different possible outputs, may be worth making this more abstract so we don't need to refactor this later.
Not sure if I want to block on this yet.